internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -283,14 +283,20 @@ func (r *ClusterDeploymentsReconciler) agentClusterInstallFinalizer(ctx context.
 			if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 				return &ctrl.Result{Requeue: true}, err
 			}
-			if err == nil {
-				if swag.StringValue(cluster.Status) == models.ClusterStatusInstalling || swag.StringValue(cluster.Status) == models.ClusterStatusPreparingForInstallation {
-					log.Infof("ClusterInstall is being deleted, cancel installation for cluster %s", *cluster.ID)
-					if _, err = r.Installer.CancelInstallationInternal(ctx, installer.V2CancelInstallationParams{
-						ClusterID: *cluster.ID,
-					}); err != nil {
-						return &ctrl.Result{Requeue: true}, err
-					}
+			// these should match the valid source states for the cancel installation transition
+			cancelStates := []string{
```

Comment:
Not for this PR, but wonder if it'd be a good idea to maintain these states in one place and then reference it across multiple files?

Reply:
Maybe we can categorize them by type `progressing / failure / preparing  / success`
And create a method for getting a list by type

not sure how many time we will  need it 

do you have other uses cases for it?

Reply:
The problem is it's not the same set in the state machine and in the "client".

We can request a cluster be canceled in any of these states but the transition is different for a cluster in `PreparingForInstallation` than for the other four so I don't think it would make sense to share here even if there was a good place for us to put this list.